src/hid_freebsd.c

```diff
@@ -100,6 +105,65 @@ copy_info(fido_dev_info_t *di, const char *path)
 	return (ok);
 }
 
+#if defined(HAS_HIDRAW)
+static int
+copy_info_hidraw(fido_dev_info_t *di, const char *path)
+{
+	int			fd = -1;
+	int			ok = -1;
+	struct usb_device_info	udi;
+	struct hidraw_devinfo	devinfo;
+	char			rawname[129];
+
+	memset(di, 0, sizeof(*di));
+	memset(&udi, 0, sizeof(udi));
+	memset(&devinfo, 0, sizeof(devinfo));
+	memset(rawname, 0, sizeof(rawname));
+
+	if ((fd = fido_hid_unix_open(path)) == -1 || is_fido(fd) == 0)
+		goto fail;
+
+	if (ioctl(fd, IOCTL_REQ(USB_GET_DEVICEINFO), &udi) == -1) {
+		if (ioctl(fd, IOCTL_REQ(HIDIOCGRAWINFO), &devinfo) == -1)
+			goto fail;
+
+		if (ioctl(fd, IOCTL_REQ(HIDIOCGRAWNAME(128)), rawname))
```

Comment:
Should we check for `ioctl()` failure `(== -1)`?

Reply:
Yes